components/kyma-environment-broker/internal/broker/broker.go

```diff
@@ -262,31 +272,25 @@ func (b *KymaEnvBroker) LastOperation(ctx context.Context, instanceID string, de
 	if err != nil {
 		return domain.LastOperation{}, errors.Wrapf(err, "while getting instance from storage")
 	}
+	_, err = url.ParseRequestURI(instance.DashboardURL)
+	if err == nil {
+		return domain.LastOperation{
+			State:       domain.Succeeded,
+			Description: "Dashboard URL already exists in the instance",
+		}, nil
+	}
 
 	status, err := b.ProvisionerClient.RuntimeOperationStatus(instance.GlobalAccountID, details.OperationData)
 	if err != nil {
-		b.Dumper.Dump("Got error: ", err)
+		b.Dumper.Dump("Provisioner client returns error on runtime operation status call: ", err)
 		return domain.LastOperation{}, errors.Wrapf(err, "while getting last operation")
 	}
 	b.Dumper.Dump("Got status:", status)
 
 	var lastOpStatus domain.LastOperationState
 	switch status.State {
 	case gqlschema.OperationStateSucceeded:
-		lastOpStatus = domain.Succeeded
-
-		// todo: this is a temporary solution until the dashboard url is saved in the director
-		b.Dumper.Dump("Saving dummy dashboard url for instance ID: ", instanceID)
-		inst, err := b.InstancesStorage.GetByID(instanceID)
-		if err != nil {
-			b.Dumper.Dump("Got error: ", err)
-		}
-		inst.DashboardURL = fixedDummyURL
-		err = b.InstancesStorage.Update(*inst)
-		if err != nil {
-			b.Dumper.Dump("Got error: ", err)
-		}
-
+		lastOpStatus = b.handleDashboardURL(instance)
```

Comment:
when the provisioner returns Success but  there is no dashboard url we get at the endpoint:
status "in progress" but description "success"